pulp_rpm/tests/functional/api/test_domains.py

```diff
@@ -0,0 +1,533 @@
+import json
+
+import pytest
+import uuid
+
+from django.conf import settings
+
+from pulpcore.client.pulp_rpm import ApiException, Copy, RpmRepositorySyncURL
+from pulpcore.client.pulpcore.exceptions import ApiException as CoreApiException
+
+from pulp_rpm.tests.functional.utils import (
+    gen_rpm_remote,
+    get_package_repo_path,
+)
+from pulp_rpm.tests.functional.constants import (
+    RPM_SIGNED_FIXTURE_URL,
+)
+
+if not settings.DOMAIN_ENABLED:
+    pytest.skip("Domains not enabled.", allow_module_level=True)
+
+
+def test_domain_create(
+    domains_api_client,
+    gen_object_with_cleanup,
+    monitor_task,
+    rpm_package_api,
+    rpm_repository_api,
+    rpm_rpmremote_api,
+):
+    """Test repo-creation in a domain."""
+    body = {
+        "name": str(uuid.uuid4()),
+        "storage_class": "pulpcore.app.models.storage.FileSystem",
+        "storage_settings": {"MEDIA_ROOT": "/var/lib/pulp/media/"},
+    }
+    domain = gen_object_with_cleanup(domains_api_client, body)
+    domain_name = domain.name
+
+    # create and sync in default domain (not specified)
+    remote = gen_object_with_cleanup(rpm_rpmremote_api, gen_rpm_remote(RPM_SIGNED_FIXTURE_URL))
+    repo_body = {"name": str(uuid.uuid4()), "remote": remote.pulp_href}
+    repo = gen_object_with_cleanup(rpm_repository_api, repo_body)
+    # Check that we can "find" the new repo in the default-domain
+    found = False
+    for repo in rpm_repository_api.list(pulp_domain="default").results:
+        found |= repo.pulp_href == repo.pulp_href
```

Comment:
Name collision on `repo`. Not sure this test is checking anything.

Reply:
IIRC, this was "a way to test list() while refetching the repo" - but I concur, it's not a useful test.